targets/TARGET_Ambiq_Micro/CMakeLists.txt

```diff
@@ -1,6 +1,8 @@
 # Copyright (c) 2020-2021 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
+include(${MBED_PATH}/tools/cmake/components.cmake)
```

Comment:
This was preexisting but, I think it's a poor architecture to to define this `MBED_PATH` variable in the application when mbed-os depends on it everywhere. CMake provides several variables to discover the current path, we should use those variables if we can. It seems very strange that we rely on the app to set this `MBED_PATH` variable, and it leads to a very confusing dependency on the app. That dependency makes building mbed-os more difficult than it should be in various cases. If the app needs to include something from mbed-os' CMake modules, it should probably define that path itself, in another variable mbed-os doesn't depend on. 

Another possible approach would be to append our module folder to [CMAKE_MODULE_PATH](https://cmake.org/cmake/help/latest/variable/CMAKE_MODULE_PATH.html) which may avoid the need to hardcode the module include paths everywhere.

Reply:
I'll fix

Reply:
My comment is more about the use of the `MBED_PATH` variable in general, rather than just in this new code. I would like to see us remove all usage of `MBED_PATH` from mbed-os, removing the dependency on the consumer defining this variable (currently we define it in the application `CMakeLists.txt` in the examples). That work should probably be a backlog item because we have roughly 280 usages of `MBED_PATH` in mbed-os at the moment.

Reply:
I agree with @rwalton-arm idea of removing the need for the app to define `MBED_PATH` -- if it is needed internally, it can be renamed something like `MBED_INTERNAL_PATH` so that people won't define the same variable by accident.

> If the app needs to include something from mbed-os' CMake modules, it should probably define that path itself, in another variable mbed-os doesn't depend on.

Yep, I thing that's the way to go.

Reply:
> My comment is more about the use of the MBED_PATH variable in general, rather than just in this new code. I would like to see us remove all usage of MBED_PATH from mbed-os, removing the dependency on the consumer defining this variable (currently we define it in the application CMakeLists.txt in the examples). That work should probably be a backlog item because we have roughly 280 usages of MBED_PATH in mbed-os at the moment.

+1, can you create an issue in this repo? THanks